### PR TITLE
Update docs structure.

### LINKED ISSUE #2692: TRE admin docs should be in its relevant section
**Describe the bug**
The using the TRE section, should be for users doing research. 
Docs relevant to the TRE Admin should be under relevant section in the docs

### PR DESCRIPTION
**Pull request by LizaShak**
## What is being addressed

Fixes #2692

Move TRE-Admin relevant docs to a relevant sections. 
Clear the Using TRE section to contain only researchers relevant info.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -126,8 +129,6 @@ nav:
                   - Overview: tre-templates/pipeline-templates/overview.md
                   - Pipeline Schema: tre-templates/pipeline-templates/pipeline-schema.md
 
-      - Deploying the AzureTRE using CI/CD: using-tre/pipelines/pipelines.md
```
Comment (marrobi) [inline mkdocs.yml:129]: I think this needs to stay here, as isn't just for quick start.

File: docs/tre-admins/upgrading-tre.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ This document will cover how AzureTRE is referenced and how to upgrade its versi
 
 ## Introduction
 
-AzureTRE referenced as an external folder in [AzureTRE deployment repository](https://github.com/microsoft/AzureTRE-Deployment). A specific version of it is downloaded as part of devcontainer setup.
+AzureTRE referenced as an external folder in [AzureTRE deployment repository](https://github.com/microsoft/AzureTRE-Deployment) (which is used as a template for your project in the quickstart guide). A specific version of AzureTRE is downloaded as part of devcontainer setup.
  A symlink is then created making it available to reference in the directory itself (it is available only for reference, any changes to it are gitignored)
 
 ## How to upgrade AzureTRE version
```
Comment (marrobi) [inline docs/tre-admins/upgrading-tre.md:10]: Does the content need to be taken from the front page of the Deployment repo and go here?

**Review by marrobi — COMMENTED** [review]
Added a couple of comments.

File: mkdocs.yml (hunk 1 above)
Comment (anastasiakuzn) [reply mkdocs.yml:129]: @LizaShak , we just moved this section technical guide. I would agree with @marrobi to keep the section as a separate sub-section under technical guide.

File: docs/tre-admins/upgrading-tre.md (hunk 2 above)
Comment (marrobi) [reply docs/tre-admins/upgrading-tre.md:10]: Don't mind either way :)

**Review by marrobi — APPROVED** [review]
LGTM.

**Pull request merged**